src/components/NavBar/MenuSwitch/SwitchItem/index.tsx

```diff
@@ -29,12 +29,40 @@ import { removeLayer } from '../../../../context/mapStateSlice';
 import { useSafeTranslation } from '../../../../i18n';
 import { clearDataset } from '../../../../context/datasetStateSlice';
 
+const appendLayerToUrl = (
+  urlLayers: string | null,
+  selectedLayers: LayerType[],
+  layer: LayerType,
+): string => {
+  const selectedLayersUrl = urlLayers !== null ? urlLayers.split(',') : [];
+
+  const filteredSelectedLayers = selectedLayers
+    .filter(l => selectedLayersUrl.includes(l.id) && l.type !== layer.type)
+    .map(l => l.id);
+
+  const updatedUrl =
+    filteredSelectedLayers.length !== 0
+      ? [...filteredSelectedLayers, layer.id]
+      : [layer.id];
+
+  return updatedUrl.join(',');
+};
+
+const removeLayerFromUrl = (
+  urlLayers: string | null,
+  layerId: string,
+): string => {
+  const selectedLayersUrl = urlLayers !== null ? urlLayers.split(',') : [];
+
+  return selectedLayersUrl.filter(l => l !== layerId).join(',');
+};
+
```

Comment:
I think we should generalize and put this in the `useUrlHistory` hook so that these functions can be reused later